flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java

```diff
@@ -1151,6 +1151,23 @@ public void coGroup(Iterable<Vertex<K, VV>> vertex, Iterable<Vertex<K, VV>> vert
 		}
 	}
 
+
+	public Graph<K, VV, EV> removeVertices(DataSet<Vertex<K, VV>> verticesToBeRemoved){
```

Comment:
Yes, this is definitely the idea, but right now you are duplicating a lot of code. Can we find a smarter way (i.e. that requires as little code duplication as possible) :)?